### DISCUSSION TITLE
Labels cannot have whitespace?

### DISCUSSION THREAD

**Discussion by astrojuanlu** [executablebooks/meta#467] [Q&A]
I was translating some reStructuredText docs, and notice that while this works:

```rst
.. _My target rst:

Explicit targets
~~~~~~~~~~~~~~~~

Reference :ref:`My target rst`.
```

the equivalent MyST doesn't:

```
(My target md)=
## Explicit targets

Reference {ref}`My target md`.
```

and Sphinx gives this `WARNING`:

```
/home/juanlu/Projects/RTD/lumache/docs/source/markdown.md:76: WARNING: undefined label: my target md
```

However, if I replace the whitespace by underscores `_`, it works.

Is this intended?

**Comment by chrisjsewell** [comment]
Heya, yeh I was unaware that RST accepted spaces, but this could be pretty easily "fixed" by altering: https://github.com/executablebooks/mdit-py-plugins/blob/a0f7ffac84b2eb768a508c2e7a1c286df0c2b56a/mdit_py_plugins/myst_blocks/index.py#L116

**Reply by astrojuanlu** [reply]
Thanks @chrisjsewell ! Should I make this a feature request then? I can open an issue in any of the repos if appropriate

**Reply by choldgraf** [reply]
I opened up https://github.com/executablebooks/mdit-py-plugins/issues/28 to track this - maybe @chrisjsewell can clean that up if I've incorrectly described something there, and @astrojuanlu can chime in too?

**Reply by chrisjsewell** [reply]
this is fixed and will be in the next release
